Account/FxAState.swift

```diff
@@ -80,36 +82,110 @@ func stateV1(fromJSON json: JSON) -> FxAState? {
                     let keyPair = RSAKeyPair(jsonRepresentation: keyPairJSON),
                     let keyPairExpiresAt = json["keyPairExpiresAt"].int64 {
-                        return CohabitingAfterKeyPairState(sessionToken: sessionToken, kA: kA, kB: kB,
+                        let kSync = FxAClient10.deriveKSync(kB)
+                        let kXCS = FxAClient10.computeClientState(kB)
+                        return CohabitingAfterKeyPairState(sessionToken: sessionToken, kSync: kSync, kXCS: kXCS,
                             keyPair: keyPair, keyPairExpiresAt: UInt64(keyPairExpiresAt))
                 }
 
             case .married:
                 if let
                     sessionToken = json["sessionToken"].string?.hexDecodedData,
-                    let kA = json["kA"].string?.hexDecodedData,
                     let kB = json["kB"].string?.hexDecodedData,
                     let keyPairJSON = json["keyPair"].dictionaryObject,
                     let keyPair = RSAKeyPair(jsonRepresentation: keyPairJSON),
                     let keyPairExpiresAt = json["keyPairExpiresAt"].int64,
                     let certificate = json["certificate"].string,
                     let certificateExpiresAt = json["certificateExpiresAt"].int64 {
-                    return MarriedState(sessionToken: sessionToken, kA: kA, kB: kB,
-                        keyPair: keyPair, keyPairExpiresAt: UInt64(keyPairExpiresAt),
-                        certificate: certificate, certificateExpiresAt: UInt64(certificateExpiresAt))
+                        let kSync = FxAClient10.deriveKSync(kB)
+                        let kXCS = FxAClient10.computeClientState(kB)
+                        return MarriedState(sessionToken: sessionToken, kSync: kSync, kXCS: kXCS,
+                                            keyPair: keyPair, keyPairExpiresAt: UInt64(keyPairExpiresAt),
+                                            certificate: certificate, certificateExpiresAt: UInt64(certificateExpiresAt))
+                }
+
+            case .separated:
+                return SeparatedState()
+
+            case .doghouse:
+                return DoghouseState()
+            }
+        }
+    }
+    return nil
+}
+
+// Same as V1 except we it contains kSync and kXCS instead of kB
+func stateV2(fromJSON json: JSON) -> FxAState? {
+    if let labelString = json["label"].string {
+        if let label = FxAStateLabel(rawValue: labelString) {
+            switch label {
+            case .engagedBeforeVerified:
+                if let
+                    sessionToken = json["sessionToken"].string?.hexDecodedData,
+                    let keyFetchToken = json["keyFetchToken"].string?.hexDecodedData,
+                    let unwrapkB = json["unwrapkB"].string?.hexDecodedData,
+                    let knownUnverifiedAt = json["knownUnverifiedAt"].int64,
+                    let lastNotifiedUserAt = json["lastNotifiedUserAt"].int64 {
```

Comment:
Looks repeated in each `case`, consider pulling them out of the case body (if possible)